comp/process/expvars/expvars.go

```diff
@@ -0,0 +1,94 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package expvars
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"go.uber.org/fx"
+
+	sysconfig "github.com/DataDog/datadog-agent/cmd/system-probe/config"
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/comp/core/log"
+	"github.com/DataDog/datadog-agent/comp/core/sysprobeconfig"
+	"github.com/DataDog/datadog-agent/comp/process/hostinfo"
+	ddconfig "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/process/runner"
+	"github.com/DataDog/datadog-agent/pkg/process/status"
+	"github.com/DataDog/datadog-agent/pkg/process/util"
+)
+
+var _ Component = (*expvarServer)(nil)
+
+type expvarServer *http.Server
+
+type dependencies struct {
+	fx.In
+
+	Lc fx.Lifecycle
+
+	Config         config.Component
+	SysProbeConfig sysprobeconfig.Component
+	HostInfo       hostinfo.Component
+
+	Log log.Component
+}
+
+func newExpvarServer(deps dependencies) (Component, error) {
+	// Initialize status
+	err := initStatus(deps)
+	if err != nil {
+		_ = deps.Log.Critical("Failed to initialize status server:", err)
+		return nil, err
+	}
+
+	expvarPort := getExpvarPort(deps)
+	expvarServer := &http.Server{Addr: fmt.Sprintf("localhost:%d", expvarPort), Handler: http.DefaultServeMux}
+
+	deps.Lc.Append(fx.Hook{
+		OnStart: func(ctx context.Context) error {
+			go func() {
+				err := expvarServer.ListenAndServe()
+				if err != nil && !errors.Is(err, http.ErrServerClosed) {
+					_ = deps.Log.Error(err)
+				}
+			}()
+			return nil
+		},
+		OnStop: expvarServer.Shutdown,
```

Comment:
Does the hook take into account potential errors thrown by the `Shutdown` function? 

I wonder if we're losing information if the server somehow fails the shutdown process.

Looking at `main_common.go`, for example, there's a log error whenever the operation fails. Otherwise we can have a `StopAPI/ExpVar()` function to wrap it up.
```golang
if err := srv.Shutdown(context.Background()); err != nil {
	log.Errorf("Error shutting down expvar server on port %v: %v", expVarPort, err)
}
```

Reply:
So that's a good point that I didn't realize. I tested this behavior myself, and noticed that what happens is `app.Stop()` throws an error, does not evaluate the rest of the stop hooks, and then returns early. I think this is probably not the behavior we want here. I changed it so that we're just logging an error instead.